static/js/reformatShortcuts.js

```diff
@@ -15,54 +15,29 @@ var SELECT_NEXT_ELEMENT     = 39; // ArrowRight
 var SELECT_PREVIOUS_ELEMENT = 37; // ArrowLeft
 var DELETE_ELEMENT          = 46; // Del
 
-var OPEN_REFORMAT_WINDOW_MESSAGE = 'open_reformat_window';
-var CLOSE_REFORMAT_WINDOW_MESSAGE = 'close_reformat_window';
-var CHANGE_ELEMENT_TYPE_MESSAGE = 'change_element_type';
-var SELECT_NEXT_ELEMENT_MESSAGE = 'select_next_element';
-var SELECT_PREVIOUS_ELEMENT_MESSAGE = 'select_previous_element';
-var DELETE_ELEMENT_MESSAGE = 'delete_element';
-var HANDLE_SHORTCUT_EVENT = 'handle_shortcut_event';
-
-// send message to reformatEventListener
-var triggerReformatEvent = function(data) {
-  var $innerDoc = utils.getPadInner().find('#innerdocbody');
-  $innerDoc.trigger(HANDLE_SHORTCUT_EVENT, data);
-}
-
 var SHORTCUT_HANDLERS = {};
 
-SHORTCUT_HANDLERS[OPEN_REFORMAT_WINDOW] = function() {
-  api.triggerEvent({ type: OPEN_REFORMAT_WINDOW_MESSAGE });
-  /*
-   * [1] by returning false, we tell the shortcutsAndMergeLinesHandler that
-   * the pressed key should NOT be canceled (preventDefault).
-   */
-  return false;
-};
+// NOTE: the functions below receive the "context" object as an argument.
 
+SHORTCUT_HANDLERS[OPEN_REFORMAT_WINDOW] = function() {
+  var reformatShortcutHandler = utils.getThisPluginProps().reformatShortcutHandler;
+  return reformatShortcutHandler.handleOpenReformatWindow();
+}
 SHORTCUT_HANDLERS[CLOSE_REFORMAT_WINDOW] = function() {
-  api.triggerEvent({ type: CLOSE_REFORMAT_WINDOW_MESSAGE });
-  return false; // [1]
-};
-
+  var reformatShortcutHandler = utils.getThisPluginProps().reformatShortcutHandler;
```

Comment:
esse cara aqui pode ser passado pra fora, não?

Reply:
Tentei fazer dessa forma no início, mas quando chega nessa linha, `reformatShortcutHandler` ainda é `undefined`.

Reply:
hum, é pq não foi definido ainda quando é chamado. ok